### PR TITLE
python3Packages.rouge-score: init at 0.1.2

### PR DESCRIPTION
**Pull request by nviets**
###### Description of changes

Adding rouge_score

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin

### CODE REVIEW

File: pkgs/development/python-modules/rouge-score/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{ lib
+, fetchPypi
+, fetchFromGitHub
+, python
+, buildPythonPackage
+, absl-py
+, nltk
+, numpy
+, six
+, pytest
```
Comment (natsukium) [inline pkgs/development/python-modules/rouge-score/default.nix:10]:
```suggestion
```

File: pkgs/development/python-modules/rouge-score/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    hash = "sha256-x9TaJoPmjJq/ATXvkV1jpGZDZm+EjlWKG59+rRf/DwQ=";
+  };
+
+  patchPhase = ''
+    cp ${testdata}/rouge/test_util.py .
+    substituteInPlace test_util.py \
+      --replace 'os.path.join(os.path.dirname(__file__), "testdata")' '${testdata}/rouge/testdata'
+  '';
+
+  nativeBuildInputs = [ absl-py nltk numpy six ];
```
Comment (natsukium) [inline pkgs/development/python-modules/rouge-score/default.nix:39]:
```suggestion
  propagatedBuildInputs = [ absl-py nltk numpy six ];
```

File: pkgs/development/python-modules/rouge-score/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  enableParallelBuilding = true;
```
Comment (natsukium) [inline pkgs/development/python-modules/rouge-score/default.nix:41]:
```suggestion
```
There is no need to build in parallel.

File: pkgs/development/python-modules/rouge-score/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  doCheck = true;
+
+  checkPhase = ''
+    runHook preCheck
+    pushd "$out"
+    ${python.interpreter} lib/python3.10/site-packages/rouge_score/test_util.py
```
Comment (natsukium) [inline pkgs/development/python-modules/rouge-score/default.nix:48]: This `test_util.py` and the one in lines 34-35 are different.
Therefore, the test does not work correctly.

**Review by natsukium — CHANGES_REQUESTED** [review]

**Comment by nviets** [comment]
@natsukium - thanks for the review. Let me know if you'd like anything else.

File: pkgs/development/python-modules/rouge-score/default.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    extension = "tar.gz";
+    hash = "sha256-x9TaJoPmjJq/ATXvkV1jpGZDZm+EjlWKG59+rRf/DwQ=";
+  };
+
+  # the tar file from pypi doesn't come with the test data
+  patchPhase = ''
```
Comment (natsukium) [inline pkgs/development/python-modules/rouge-score/default.nix:34]: It is better to patch with `postPatch` than to override the `patchPhase` behavior, I think.

**Comment by natsukium** [comment]
Then, could you change the commit message?
Otherwise, it seems fine.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 229208` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.rouge-score</li>
    <li>python310Packages.rouge-score.dist</li>
    <li>python311Packages.rouge-score</li>
    <li>python311Packages.rouge-score.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]

File: pkgs/development/python-modules/rouge-score/default.nix [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    hash = "sha256-ojqk6U2caS7Xz4iGUC9aQVHrKb2QNvMlPuQAL/jJat0=";
+  };
+in buildPythonPackage rec {
+  pname = "rouge_score";
+  version = "0.1.2";
+  format = "setuptools";
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rouge-score/default.nix:28]:
```suggestion
    pname = "rouge_score";
    inherit version;
```

File: pkgs/development/python-modules/rouge-score/default.nix [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    extension = "tar.gz";
+    hash = "sha256-x9TaJoPmjJq/ATXvkV1jpGZDZm+EjlWKG59+rRf/DwQ=";
+  };
+
+  # the tar file from pypi doesn't come with the test data
+  postPatch = ''
+    substituteInPlace rouge_score/test_util.py \
+      --replace 'os.path.join(os.path.dirname(__file__), "testdata")' '"${testdata}/rouge/testdata/"'
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rouge-score/default.nix:37]: Can we avoid this by fetching from github?

Comment (nviets) [reply pkgs/development/python-modules/rouge-score/default.nix:38]: Only if we disable nearly all the checks. The test data doesn't come with the tarball from pypi. The test data isn't very large, fortunately.

**Comment by nviets** [comment]
Thanks for the cleanup, @SuperSandro2000! Need anything else from me?

**Pull request merged**